indexer/services/roundtable/src/tasks/aggregate-trading-rewards.ts

```diff
@@ -213,23 +229,305 @@ export class AggregateTradingReward {
+        },
+      ).concat(otherAggregationUpdateObjects),
+    };
+  }
+
+  private getEndedAt(interval: Interval): IsoString {
+    const startedAt: IsoString = this.getStartedAt(interval);
+
+    return DateTime
+      .fromISO(startedAt, UTC_OPTIONS)
+      .endOf(this.getDateTimeUnit())
+      .plus({ milliseconds: 1 })
+      .toISO();
+  }
+
+  private getDateTimeUnit(): DateTimeUnit {
+    switch (this.period) {
+      case TradingRewardAggregationPeriod.DAILY:
+        return DateTimeUnit.day;
+      case TradingRewardAggregationPeriod.WEEKLY:
+        return DateTimeUnit.week;
+      case TradingRewardAggregationPeriod.MONTHLY:
+        return DateTimeUnit.week;
+      default:
+        throw new Error(`Invalid period ${this.period}`);
+    }
+  }
+
+  private async setAggregationUpdateAndCreateObjects(
+    aggregationUpdateAndCreateObjects: AggregationUpdateAndCreateObjects,
+  ): Promise<void> {
+    const createObjectsChunks: TradingRewardAggregationCreateObject[][] = _.chunk(
+      aggregationUpdateAndCreateObjects.createObjects,
+      config.AGGREGATE_TRADING_REWARDS_CHUNK_SIZE,
+    );
+    for (const createObjectsChunk of createObjectsChunks) {
+      logger.info({
+        at: 'aggregate-trading-rewards#setAggregationUpdateAndCreateObjects',
+        message: 'Creating trading reward aggregations',
+        count: createObjectsChunk.length,
+        createObjectsChunk: JSON.stringify(createObjectsChunk),
+      });
+      await Promise.all(_.map(
+        createObjectsChunk,
+        (createObject: TradingRewardAggregationCreateObject) => {
+          return TradingRewardAggregationTable.create(createObject);
+        },
+      ));
+      logger.info({
+        at: 'aggregate-trading-rewards#setAggregationUpdateAndCreateObjects',
+        message: 'Created trading reward aggregations',
+        count: createObjectsChunk.length,
+      });
+    }
+
+    const updateObjectsChunks: TradingRewardAggregationUpdateObject[][] = _.chunk(
+      aggregationUpdateAndCreateObjects.updateObjects,
+      config.AGGREGATE_TRADING_REWARDS_CHUNK_SIZE,
+    );
+    for (const updateObjectsChunk of updateObjectsChunks) {
+      logger.info({
+        at: 'aggregate-trading-rewards#setAggregationUpdateAndCreateObjects',
+        message: 'Updating trading reward aggregations',
+        count: updateObjectsChunk.length,
+        updateObjectsChunk: JSON.stringify(updateObjectsChunk),
+      });
+      await Promise.all(_.map(
+        updateObjectsChunk,
+        (updateObject: TradingRewardAggregationUpdateObject) => {
+          return TradingRewardAggregationTable.update(updateObject);
+        },
+      ));
+      logger.info({
+        at: 'aggregate-trading-rewards#setAggregationUpdateAndCreateObjects',
+        message: 'Updated trading reward aggregations',
+        count: updateObjectsChunk.length,
+      });
+    }
+  }
+
+  private async setProcessedTime(processedTime: IsoString): Promise<void> {
+    logger.info({
```

Comment:
should we make this a stat instead?

Reply:
This is just for debugging, I'm not tracking the latency of this call